core/lib/types/src/commitment/mod.rs

```diff
@@ -327,20 +330,32 @@ impl L1BatchAuxiliaryOutput {
                 system_logs,
                 state_diffs,
                 aux_commitments,
+                blob_commitments,
             } => {
+                // TODO (PLA-815): Remove `IGNORE_1_4_2_UPGRADE`.
+                let protocol_version = if std::env::var("IGNORE_1_4_2_UPGRADE").is_ok() {
```

Comment:
should we just switch era contracts?

Reply:
We can't update era-contracts currently, we either need this kludge so upgrade test passes or merge these changes in the final PR with contracts updated. I think first is better

Reply:
ok, I'll try to fix it with a new tool ;)

Reply:
Sorry for unresolving. Can we make this kludge a bit more verbose by 
1. Changing the env var to `IGNORE_1_4_2_UPGRADE_FOR_UPGRADE_TEST`?
2. Add an explanation comment: "// This is used only in upgrade test, because our system isn't flexible enough right now to support WIP changes to contracts" or something like this. Issues in linear are getting removed after a while, so we may forget it and then keep wondering whether it's needed for some long forgotten tooling.
3. Add a `tracing::error!("Ignoring the 1.4.2 upgrade");` just in case.

Reply:
please mention it in the comment in the code above.